cpp/src/DataStorm/DataElementI.cpp

```diff
@@ -178,7 +189,7 @@ DataElementI::attach(
         session->subscriberInitialized(topicId, id > 0 ? data.id : -data.id, data.samples, key, shared_from_this());
     if (!samplesI.empty())
     {
-        return [=, self = shared_from_this()]()
+        return [=, samplesI = std::move(samplesI), self = shared_from_this()]()
```

Comment:
My understanding is that the previous code, captures `samplesI` as a copy, which is them passed by reference to initSamples.

I updated it to move capture it because it was not used anywhere else, but maybe is simpler to just capture it by reference.